File: carpet_cleaner.controller/main/java/kz/nu/carpet_cleaner/controller/model/OrderFullRecord.java
package kz.nu.carpet_cleaner.controller.model;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import static kz.nu.carpet_cleaner.controller.util.ServerUtil.nvl;

public class OrderFullRecord {
  public String id;
  public String number;
  public String name;
  public String surname;
  public String patronymic;
  public String customerFio;
  public String customerPhoneNumber;
  public String pickUpAddress;
  public String pickUpDateDisplay;
  public Date pickUpDate;
  public List<MerchantDisplayRecord> merchantList = new ArrayList<>();

  public void formatData() {
    this.customerFio = nvl(this.surname, false) + nvl(this.name, false) + nvl(this.patronymic,true);
    if(this.pickUpDate!=null) this.pickUpDateDisplay = new SimpleDateFormat("dd.MM.yyyy").format(this.pickUpDate);
  }
}

File: carpet_cleaner.mobile/src/app/models/OrderFullRecord.ts
import {MerchantRecord} from "./MerchantRecord";

export class OrderFullRecord {
  id:string;
  number:string;
  customerFio:string;
  customerPhoneNumber:string;
  pickUpAddress:string;
  pickUpDateDisplay:string;
  price:number;
  merchantList:MerchantRecord[];

  constructor() {

  }
}